src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -1253,52 +1253,165 @@ class C
+            Diagnostic(ErrorCode.WRN_ObsoleteMembersShouldNotBeRequired, "Prop1").WithArguments("S.Prop1").WithLocation(8, 25)
+        );
+    }
+
+    [Fact]
+    public void ObsoleteMember_ObsoleteContext()
+    {
+        var comp = CreateCompilationWithRequiredMembers("""
+            using System;
+            #pragma warning disable CS0649 // Unassigned field
+            [Obsolete]
+            class C
+            {
+                [Obsolete]
+                public required int Field;
+                [Obsolete]
+                public required int Prop1 { get; set; }
+            }
+            """);
+
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void ObsoleteMember_ObsoleteOrSetsRequiredMembersConstructors_01()
+    {
+        var comp = CreateCompilationWithRequiredMembers("""
+            using System;
+            using System.Diagnostics.CodeAnalysis;
+            #pragma warning disable CS0649 // Unassigned field
+            class C
+            {
+                [Obsolete]
+                public C() { }
+                [SetsRequiredMembers]
+                public C(int i) { }
+
+                [Obsolete]
+                public required int Field;
+                [Obsolete]
+                public required int Prop1 { get; set; }
+            }
+
+            """);
 
-        // PROTOTYPE(req): Confirm with LDM whether we want a warning here.
         comp.VerifyDiagnostics();
     }
 
+    [Fact]
+    public void ObsoleteMember_ObsoleteOrSetsRequiredMembersConstructors_02()
+    {
+        var comp = CreateCompilationWithRequiredMembers("""
+            using System;
+            using System.Diagnostics.CodeAnalysis;
+            #pragma warning disable CS0649 // Unassigned field
+            class C
+            {
+                [Obsolete]
+                public C() { }
+                [SetsRequiredMembers]
+                public C(int i) { }
+
+                public C(bool b) { }
+
+                [Obsolete]
+                public required int Field;
+                [Obsolete]
+                public required int Prop1 { get; set; }
+            }
+
+            """);
+
+        comp.VerifyDiagnostics(
+            // (14,25): warning CS9513: Required member 'C.Field' should not be attributed with 'ObsoleteAttribute' unless the containing type is obsolete or all constructors are obsolete.
+            //     public required int Field;
+            Diagnostic(ErrorCode.WRN_ObsoleteMembersShouldNotBeRequired, "Field").WithArguments("C.Field").WithLocation(14, 25),
+            // (16,25): warning CS9513: Required member 'C.Prop1' should not be attributed with 'ObsoleteAttribute' unless the containing type is obsolete or all constructors are obsolete.
+            //     public required int Prop1 { get; set; }
+            Diagnostic(ErrorCode.WRN_ObsoleteMembersShouldNotBeRequired, "Prop1").WithArguments("C.Prop1").WithLocation(16, 25)
+        );
+    }
```

Comment:
The spec update calls out for property without setter/initer or with less accessible setter/initer. Do we have tests covering those?

Reply:
https://github.com/dotnet/roslyn/pull/61436/files#diff-edd6e15a777d20344e841412cb37cad70aa68691075fc9d77fa12b630377e2c9R1262-R1285

Reply:
Thanks. I see that covers `set` with various accessibility. Do we also have missing setter/initer? (I searched but couldn't find it)

Reply:
That same test covers that as well.

Reply:
Cool. Thanks!